en/docs/install-and-setup/setup/kubernetes-operators/k8s-api-operator/manage-integration/integration-deployments.md

```diff
@@ -450,4 +481,170 @@ Use the following methods to invoke the inbound endpoints in HTTP and HTTPS tran
-    ```    
\ No newline at end of file
+    ```    
+
+
+## Managing resources for pods in EI deployment
+
+When creating a deployment operator allows to define required resources to run pods created by that deployment. In the same manner, you can define resource limits as well. Following example shows how to set CPU and memory resources at `integration_cr.yaml` file.
+
+```yaml
+apiVersion: wso2.com/v1alpha2
+kind: Integration
+metadata:
+  name: test-integration
+spec:
+  image: <MI based Docker image for integration>
+  deploySpec:
+    minReplicas: 1
+    requestCPU: "500m"
+    reqMemory: "512Mi"
+    cpuLimit: "2000m"
+    memoryLimit: "2048Mi"
+```
+
+You can set the default resource configs at integration_controller_conf.yaml and add them to the configMap. Whenever they are not defined at `integration_cr.yaml` file, operator will pick defaults from the configMap. 
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: integration-config
+data:
+  requestCPU: "500m"
+  reqMemory: "512Mi"
+  cpuLimit: "2000m"
+  memoryLimit: "2048Mi"
+```
+
+Please refer to [Managing Resources for Containers](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/) to learn more. 
+
+
+## Enable Auto Scaling for deployment
+
+When traffic is increased your pods may need to horizontally scaled. Kubernetes provides a way to define resource limits and a policy for the deployment so that it will auto scale the deployment looking at resource usage. Please read on [Horizontal Pod Scaling (HPA)](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) in Kubernetes.
+
+You can enable scaling and define maximum number of replicas to scale under `autoScale` of `integration_cr.yaml` as below. 
+
+  ```yaml
+apiVersion: wso2.com/v1alpha2
+kind: Integration
+metadata:
+  name: test-integration
+spec:
+  autoScale:
+    enabled: "true"
+    maxReplicas: 3
+  ```
+
+You can enable auto scaling by default by defining it at `integration_controller_conf.yaml` file. These values will get overridden by what is defined at  `integration_cr.yaml` file. 
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: integration-config
+data:
+  enableAutoScale: "true"
+  maxReplicas: "5"
+```
+
+Note how you can set required resources and resource limits for the pods in the deployment referring to the above section [Managing resources for pods in EI deployment](#Managing-resources-for-pods-in-EI-deployment). HPA configs are injected through the integration configmap. See how you can define it at [integration_controller_conf.yaml]([integration_controller_conf.yaml](https://github.com/wso2/k8s-api-operator/blob/master/api-operator/deploy/controller-configs/integration_controller_conf.yaml)) file. 
+
+```yaml
+  hpaMetrics: |
+    - type: Resource
+      resource:
+        name: cpu
+        target:
+          type: Utilization
+          averageUtilization: 50
```

Comment:
When the CPU utilisation for one pod exceeds this value, the scaling starts? Is that how it works?

@abeykoon

Reply:
There is an algorithm. Quoting K8s docs, 

```
For per-pod resource metrics (like CPU), the controller fetches the metrics from the resource metrics API for each Pod targeted by the HorizontalPodAutoscaler. Then, if a target utilization value is set, the controller calculates the utilization value as a percentage of the equivalent resource request on the containers in each Pod. If a target raw value is set, the raw metric values are used directly. The controller then takes the mean of the utilization or the raw value (depending on the type of target specified) across all targeted Pods, and produces a ratio used to scale the number of desired replicas.
```